google-datacatalog-kafka-connector/tools/cleanup_datacatalog.py

```diff
@@ -0,0 +1,120 @@
+                                                query=query,
+                                                order_by='relevance',
+                                                page_size=1000)
+    datacatalog_entry_name_pattern = '(?P<entry_group_name>.+?)/entries/(.+?)'
+
+    entry_group_names = []
+    for result in search_results:
+        try:
+            datacatalog.delete_entry(result.relative_resource_name)
+            logging.info('Entry deleted: %s', result.relative_resource_name)
+            entry_group_name = re.match(
+                pattern=datacatalog_entry_name_pattern,
+                string=result.relative_resource_name).group('entry_group_name')
+            entry_group_names.append(entry_group_name)
+        except exceptions.GoogleAPICallError as e:
+            logging.warning('Exception deleting entry: %s', str(e))
+
+    # Delete any pre-existing Entry Groups.
+    for entry_group_name in set(entry_group_names):
+        try:
+            datacatalog.delete_entry_group(entry_group_name)
+            logging.info('--> Entry Group deleted: %s', entry_group_name)
+        except exceptions.GoogleAPICallError as e:
+            logging.warning('Exception deleting entry group: %s', str(e))
+
+
+def __delete_tag_templates(project_id, location_id):
+    tag_template_id = 'kafka_cluster_metadata'
+
+    try:
+        datacatalog.delete_tag_template(
+            datacatalog_v1beta1.DataCatalogClient.tag_template_path(
+                project=project_id,
+                location=location_id,
+                tag_template=tag_template_id),
+            force=True)
+        logging.info('--> Tag Template deleted: %s', tag_template_id)
+    except exceptions.GoogleAPICallError as e:
+        logging.warning('Exception deleting Tag Template: %s', str(e))
+
+    tag_template_id = 'kafka_topic_metadata'
+
+    try:
+        datacatalog.delete_tag_template(
+            name=datacatalog_v1beta1.DataCatalogClient.tag_template_path(
+                project=project_id,
+                location=location_id,
+                tag_template=tag_template_id),
+            force=True)
+        logging.info('--> Tag Template deleted: %s', tag_template_id)
+    except exceptions.GoogleAPICallError as e:
+        logging.warning('Exception deleting Tag Template: %s', str(e))
+
+
+def __parse_args():
+    parser = argparse.ArgumentParser(description='Command line to clean up all'
+                                     ' Kafka metadata on Datacatalog')
+
+    parser.add_argument(
+        '--datacatalog-project-ids',
+        help='List of Google Cloud project IDs split by comma, '
+        'At least one must be specified',
+        required=True)
+    parser.add_argument(
+        '--datacatalog-location-id',
+        help='Location id which is the Region that your Datacatalog resides',
+        default='us-central1')
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
```

Comment:
Why do we need this? If it is used only in tests perhaps it should be in test directory, I'm not sure if we want to provide the user with a tool that deletes all their data

Reply:
It's for developers - we assume that the connector is a sample code that could be extended or adapted based on users' needs. In README it is described how to use this cleanup.
Since all the connectors have similar structure and all of them have a cleanup script in tools/, I did the same here for Kafka.